File: backend/travel-planner-api/TravelPlannerApi/Models/Trips/TripDayActivity.cs
namespace TravelPlannerApi.Models.Trips;

public class TripDayActivity
{
    public required string Id { get; set; }

    public required string Address { get; set; }

    public required double Latitude { get; set; }

    public required double Longitude { get; set; }

    public string? Name { get; set; }

    public string? Description { get; set; }

    public string? ImageUrl { get; set; }

    public TimeSpan? Duration { get; set; }
}

File: frontend/travel-planner-webapp/src/services/trips/TripDayActivity.ts
import { Duration } from 'moment';


export interface TripDayActivity {
    readonly id: string;
    readonly address: string;
    readonly latitude: number;
    readonly longitude: number;
    readonly name: string | null;
    readonly description: string | null;
    readonly imageUrl: string | null;
    readonly duration: Duration | null;
}